utils/errors/errors.go

```diff
@@ -145,26 +155,31 @@ func NewWithSkip(msg string, skip int) ErrorExt {
 
 //NewWithSkipAndStatus creates a new error skipping the number of function on the stack and GRPC status
 func NewWithSkipAndStatus(msg string, skip int, status *grpcstatus.Status) ErrorExt {
-	return WrapWithSkipAndStatus(fmt.Errorf(msg), "", skip+1, status)
+	return wrapWithSkipAndStatus(fmt.Errorf(msg), "", skip+1, status)
 }
 
 //Wrap wraps an existing error and appends stack information if it does not exists
 func Wrap(err error, msg string) ErrorExt {
 	return WrapWithSkip(err, msg, 1)
 }
 
-//Wrap wraps an existing error and appends stack information if it does not exists along with GRPC status
+//WrapWithRawData wraps an existing error and appends stack information if it does not exists
+func WrapWithRawData(err error, msg string, rawData ...interface{}) ErrorExt {
```

Comment:
why create new func only for raw data instead of adding for each Wrap* functions? rawData is accepting all types of args, what if we need to add new args as interface....for example,

```
func Wrap(err error, msg string, rawData ...interface{}) ..
// become
func Wrap(err error, msg string, anything interface{}, rawData ...interface{})
```

in case of it, I would prefer create new function to prevent potential breaking change in the future.

Reply:
By this reasoning we should not be modifying the `.New()` function either.

Reply:
I think we should add another `NewWithRawData` function as well.. thoughts?

Reply:
it needs another func `WrapWithSkipAndRawData`, I have no strong opinion on this, for now I feel it's good enough because in most of cases we wrap data for returned error rather than creating an new error with raw data